portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,9 +33,34 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     messages.add("first message");
     messages.add("second message");
     messages.add("third message");
-    Gson gson = new Gson();
-    String jsonMessage = gson.toJson(messages);
+    String jsonMessage = convertToJson(messages);
+    response.setContentType("application/json;");
+    response.getWriter().println(jsonMessage);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String email = getParameter(request, "email", null);
+    String name = getParameter(request, "name", null);
+    String message = getParameter(request, "message", null);
+    Message messageObj = new Message(name, email, message);
```

Comment:
Is it possible that someone submits the form with all fields empty? Would you still want to create a message object then?